### PR TITLE
allow custom request headers

### PR DESCRIPTION
**Pull request by jihuang-adobe**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by ghenzler** [comment]
@jihuang-adobe What is your use case that requires to set an additional header? Do you have some intermediate infrastructure that requires that header? 

(I'm asking because potentially it could be better to explicitly provide the functionality over this generic approach, but it depends on if it is very specific to your setup or not)

**Comment by jihuang-adobe** [comment]
Hi @ghenzler.  I am currently have AEM graphql endpoint behind a proxy.  The proxy can potentially take a list of custom headers, so I thought it would be convenient if the AEM Headless Client would accept custom headers as an option.

**Review by easingthemes — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 1]
```diff
@@ -146,6 +148,13 @@ class AEMHeadless {
       }
     }
 
+    if (this.headers) {
+      requestOptions.headers = {
+        ...requestOptions.headers,
+        ...this.headers
```
Comment (easingthemes) [inline src/index.js:154]: Better to reverse order, to allow single request to override global headers

**Review by jihuang-adobe — COMMENTED** [review]
updated according to feedback

**Pull request merged**